model/project_ref.go

```diff
@@ -800,30 +852,19 @@ func FindMergedProjectRef(identifier string, version string, includeProjectConfi
 	return pRef, nil
 }
 
-// GetProjectRefMergedWithRepo merges the project with the repo that matches it, if one exists.
+// GetProjectRefMergedWithRepo merges the project with the repo, if one exists.
 // Otherwise, it will return the project as given.
 func GetProjectRefMergedWithRepo(pRef ProjectRef) (*ProjectRef, error) {
 	if !pRef.UseRepoSettings() {
 		return &pRef, nil
 	}
-	if pRef.RepoRefId != "" {
```

Comment:
If useResoSettings() is true then RepoRefId is also true. I couldn't actually find a case where we relied on the FindRepoRefByOwnerAndRepo case being used.